src/meili/match_route.cc

```diff
@@ -208,69 +216,145 @@ bool MergeRoute(std::vector<EdgeSegment>& route, const State& source, const Stat
     throw std::logic_error("The first edge must be an origin (invalid predecessor)");
   }
 
-  MergeEdgeSegments(route, segments.rbegin(), segments.rend());
+  // MergeEdgeSegments(route, segments.rbegin(), segments.rend());
+  route.insert(route.end(), segments.crbegin(), segments.crend());
   return true;
 }
 
-std::vector<EdgeSegment> MergeRoute(const State& source, const State& target) {
-  std::vector<EdgeSegment> route;
-  MergeRoute(route, source, target);
-  return route;
+void cut_segments(const std::vector<MatchResult>& match_results,
```

Comment:
here we cut edges on the path where there were interpolated traceponts that were set at breakpoints. TODO add a comment to the top of this method to say that that is the purpose of this method.